app/views/courses/show.html.erb

```diff
@@ -0,0 +1,28 @@
+<h1><%= @course.display_name %></h1>
+<hr>
+<% if current_user.nil? %>
+<div>You need to sign in to be able to join this class.</div>
+<br>
+<div class="sign-in-panel">
+<% if Rails.env == "development" %>
+<%= link_to "Developer Login", "/home/developer_login" %><br />
+<% end %>
+<%= render "devise/shared/links" %>
```

Comment:
Is this the only place where we're doing this?  If so, it seems like a lot of weird, non-obvious code to achieve what a single link to a sign-in page would do.